tests/tests.js

```diff
@@ -151,32 +151,43 @@ var runTests = function runTests(arrayFrom) {
 	});
 
 	test('works with a mapping function', function (t) {
-		var original = [1, 2, 3];
-		var actual = arrayFrom(original, function (value, index) {
-			t.equal(value, original[index], 'value and index are correct');
-			t.equal(arguments.length, 2, 'value and index are only arguments passed to the mapping function');
-			return value * 2;
+		t.test('with arrays', function (te) {
+			var original = [1, 2, 3];
+			var actual = arrayFrom(original, function (value, index) {
+				t.equal(value, original[index], 'value and index are correct');
+				t.equal(arguments.length, 2, 'value and index are only arguments passed to the mapping function');
+				return value * 2;
+			});
+			t.deepEqual(actual, [2, 4, 6]);
+			te.end();
+		});
+
+		t.test('with strings', function (te) {
+			var actual = arrayFrom('abc', function (c) {
+				return c.toUpperCase();
+			});
+			t.deepEqual(actual, ['A', 'B', 'C']);
+			te.end();
 		});
-		t.deepEqual(actual, [2, 4, 6]);
 
 		t.test('accepts an object thisArg', function (st) {
 			var context = {};
-			arrayFrom(original, function () {
+			arrayFrom([0], function () {
```

Comment:
can we leave these as-is, and define `var original` in the scope between lines 153 and 154 instead?